File: tests/average_true_range_bands_test.py
import numpy as np
from taflow import AverageTrueRangeBands
def test_average_true_range_bands_lifecycle():
    s=AverageTrueRangeBands(np.array([],float),np.array([],float),np.array([],float),2,2);s.extend([2,2],[1,1],[1.5,1.5]);assert s.value is not None;s.reset();assert len(s)==0

File: crates/taflow-core/src/indicators/average_true_range_bands_test.rs
use super::average_true_range_bands::AverageTrueRangeBands;
#[test]
fn lifecycle() {
    let mut s = AverageTrueRangeBands::new(2, 2.0).unwrap();
    s.append(2.0, 1.0, 1.5);
    assert!(s.append(2.0, 1.0, 1.5).is_none());
    assert!(s.append(2.0, 1.0, 1.5).is_some());
    s.reset();
    assert!(s.value().is_none());
}
